libraries/Microsoft.Bot.StreamingExtensions/IContentStream.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Implemented by stream attachments compatible with the Bot Framework Protocol 3 with Streaming Extensions.
+    /// </summary>
+    public interface IContentStream
+    {
+        /// <summary>
+        /// Gets a guid to use as the unique identifier of this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Guid.
+        /// </value>
+        Guid Id { get; }
+
+        /// <summary>
+        /// Gets or sets the name of the type of the object contained within this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Plain text type name.
+        /// </value>
+        string Type { get; set; }
+
+        /// <summary>
+        /// Gets or sets the length of this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Null or a numeric value bound by Min/Max Int.
+        /// </value>
+        int? Length { get; set; }
+
```

Comment:
Why is this nullable? Remove.

Reply:
We need a way of knowing we don't know the length of a stream.